src/core/dev_api/openvino/runtime/string_aligned_buffer.hpp

```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/runtime/aligned_buffer.hpp"
+
+namespace ov {
+
+/// \brief StringAlignedBuffer class to store pointer to pre-allocated buffer with std::string objects
+/// it is responsible for deallocation of std::string objects that will be stored in the buffer
+class StringAlignedBuffer : public ov::AlignedBuffer {
+public:
+    StringAlignedBuffer();
+    StringAlignedBuffer(size_t num_elements, size_t byte_size, size_t alignment, bool initialize);
+
+    StringAlignedBuffer(StringAlignedBuffer&& other);
+    StringAlignedBuffer& operator=(StringAlignedBuffer&& other);
+
+    virtual ~StringAlignedBuffer();
+
+private:
+    StringAlignedBuffer(const StringAlignedBuffer&) = delete;
+    StringAlignedBuffer& operator=(const StringAlignedBuffer&) = delete;
+
+protected:
+    size_t m_num_elements;
```

Comment:
why do we need it?
We can calculate `m_num_elements` as `byte_size / ov::element_type::string.size()`

Reply:
Or maybe someone will use `StringAlignBuffer` instead of `AlignBuffer` and allocate memory bigger than needed. Also, I worried about that logic of computing number of strgin objects to be aligned with logic in Constant.